types/wlr_linux_dmabuf_v1.c

```diff
@@ -521,32 +518,24 @@ static void feedback_tranche_send(
 
 	zwp_linux_dmabuf_feedback_v1_send_tranche_flags(resource, tranche->flags);
 
-	for (size_t i = 0; i < tranche->formats.len; i++) {
-		const struct wlr_drm_format *fmt = tranche->formats.formats[i];
-
-		// We always support buffers with an implicit modifier
-		zwp_linux_dmabuf_feedback_v1_send_tranche_modifier(resource, fmt->format,
-			DRM_FORMAT_MOD_INVALID >> 32, DRM_FORMAT_MOD_INVALID & 0xFFFFFFFF);
-
-		for (size_t j = 0; j < fmt->len; j++) {
-			uint32_t modifier_lo = fmt->modifiers[j] & 0xFFFFFFFF;
-			uint32_t modifier_hi = fmt->modifiers[j] >> 32;
-			zwp_linux_dmabuf_feedback_v1_send_tranche_modifier(resource,
-				fmt->format, modifier_hi, modifier_lo);
-		}
-	}
+    zwp_linux_dmabuf_feedback_v1_send_tranche_formats(resource, &tranche->format_indices);
 
 	zwp_linux_dmabuf_feedback_v1_send_tranche_done(resource);
 }
 
-static void feedback_send(const struct wlr_linux_dmabuf_feedback_v1 *feedback,
+static void feedback_send(struct wlr_linux_dmabuf_v1 *linux_dmabuf,
+        const struct wlr_linux_dmabuf_feedback_v1 *feedback,
 		struct wl_resource *resource) {
 	struct wl_array dev_array = {
 		.size = sizeof(feedback->main_device),
 		.data = (void *)&feedback->main_device,
 	};
 	zwp_linux_dmabuf_feedback_v1_send_main_device(resource, &dev_array);
 
+    zwp_linux_dmabuf_feedback_v1_send_format_table(resource,
+            linux_dmabuf->format_table.fd,
```

Comment:
This sends the FD in read-write mode. This means clients can write to the format table, and confuse other clients.